presto-main/src/main/java/com/facebook/presto/cost/SymbolStatsEstimate.java

```diff
@@ -0,0 +1,185 @@
+/*
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package com.facebook.presto.cost;
+
+import java.util.Objects;
+
+import static com.google.common.base.MoreObjects.toStringHelper;
+import static com.google.common.base.Preconditions.checkArgument;
+import static java.lang.Double.NaN;
+import static java.lang.Double.isNaN;
+import static java.lang.String.format;
+
+public class SymbolStatsEstimate
+{
+    public static final SymbolStatsEstimate UNKNOWN_STATS = SymbolStatsEstimate.builder().build();
+
+    // for now we support only types which map to real domain naturally and keep low/high value as double in stats.
+    private final double lowValue;
+    private final double highValue;
+    private final double nullsFraction;
+    private final double averageRowSize;
+    private final double distinctValuesCount;
+
+    public SymbolStatsEstimate(double lowValue, double highValue, double nullsFraction, double averageRowSize, double distinctValuesCount)
+    {
+        checkArgument(
+                lowValue <= highValue || (isNaN(lowValue) && isNaN(highValue)),
+                "low value must be less than or equal to high value or both values have to be NaN, got %s and %s respectively",
+                lowValue,
+                highValue);
+        this.lowValue = lowValue;
+        this.highValue = highValue;
+
+        checkArgument((0 <= nullsFraction && nullsFraction <= 1.) || isNaN(nullsFraction), "Nulls fraction should be within [0, 1] or NaN, got: %s", nullsFraction);
+        this.nullsFraction = nullsFraction;
+
+        checkArgument(averageRowSize >= 0 || isNaN(averageRowSize), "Average row size should be non-negative or NaN, got: %s", averageRowSize);
+        this.averageRowSize = averageRowSize;
+
+        checkArgument(distinctValuesCount >= 0 || isNaN(distinctValuesCount), "Distinct values count should be non-negative, got: %s", distinctValuesCount);
+        this.distinctValuesCount = distinctValuesCount;
+    }
+
+    public double getLowValue()
+    {
+        return lowValue;
+    }
+
+    public double getHighValue()
+    {
+        return highValue;
+    }
+
+    public boolean isRangeEmpty()
+    {
+        return isNaN(lowValue) && isNaN(highValue);
+    }
+
+    public double getNullsFraction()
+    {
+        if (isRangeEmpty()) {
```

Comment:
it is weird that this is the only case where `isRangeEmpty` is checked. What about values fraction, distinct values count, average size?